### PR TITLE
Fix firewall rule products support which was missing handling

### PR DESCRIPTION
**Pull request by filipowm**
Resource `cloudflare_firewall_rule` contained action `bypass` and field `products`, however there was no handling implemented what was resulting in error:
```
products must be specified for the 'bypass' action
```

### CODE REVIEW

File: cloudflare/resource_cloudflare_firewall_rule_test.go [hunk 1]
```diff
@@ -84,3 +84,49 @@ func testFirewallRuleConfig(resourceID, zoneID, paused, description, expression,
 		}
 		`, resourceID, zoneID, paused, description, expression, action, priority)
 }
+
+func TestFirewallRuleBypass(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := "cloudflare_firewall_rule." + rnd
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+
+	filterQuoted := `(http.host eq \"` + rnd + `\")`
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_firewall_rule_test.go]: this rule is invalid but i'm unsure why it's not failing when creating it in the tests (as the rule is parsed for validity on creation).

```suggestion
	filterQuoted := `(http.host eq \"terraform.cfapi.net\")`
```

(or pull it from the env using `CLOUDFLARE_DOMAIN` if you'd rather)

Comment (filipowm) [reply cloudflare/resource_cloudflare_firewall_rule_test.go]: I checked on CF and any string works.

But I've changed it to proposed `CLOUDFLARE_DOMAIN`,

**Comment by jacobbednarz** [comment]
Thanking you @filipowm 🍰

**Comment by thommy101** [comment]
Hi, any chance this will be fixed in a new patch release any time soon? We would love to use this feature.

**Comment by filipowm** [comment]
@jacobbednarz Any idea when will be the next release?

**Comment by patryk** [comment]
We usually release ~ once a month, but it's not a fixed schedule. I'll see about releasing today.

**Pull request merged**